src/extension.ts

```diff
@@ -111,41 +126,35 @@ export async function activate(context: vscode.ExtensionContext): Promise<SFCAEx
         return codeAnalyzerRunner.runAndDisplay(Constants.COMMAND_RUN_ON_SELECTED, targetStrings);
     });
 
-    onDidSaveTextDocument(async (textDocument: vscode.TextDocument) => {
-        if (settingsManager.getAnalyzeOnSave()) {
-            await codeAnalyzerRunner.runAndDisplay(Constants.COMMAND_RUN_ON_ACTIVE_FILE, [textDocument.fileName]);
-        }
-    });
-
-    onDidOpenTextDocument(async (textDocument: vscode.TextDocument) => {
-        if (settingsManager.getAnalyzeOnOpen()) {
-            if (_isValidFileForAnalysis(textDocument.uri)) {
-                await codeAnalyzerRunner.runAndDisplay(Constants.COMMAND_RUN_ON_ACTIVE_FILE, [textDocument.fileName]);
-            }
-        }
-    });
-
 
     // =================================================================================================================
     // ==  Diagnostic Management Functionality
     // =================================================================================================================
+
+    // TODO: We should look to see if we can make these commands appear conditionally upon whether we have diagnostics
```

Comment:
Pretty sure there's a way to do this. Let's create a work item for looking into it?